value-streams-as-code.md

```diff
@@ -0,0 +1,106 @@
+
+### The **helloapp-nodejs** Application
+
+The **helloapp-nodejs** application is a simple NodeJS app that serves a simple index page with a message. 
+
+If you navigate to **helloapp-nodejs** in your Jenkins master, you'll notice that there are two stages, a `Web Tests` stage used to test the NodeJS app using a selenium sandbox which runs on the `development` branch and a `Build and Push Image` stage that runs on `master`. <p><img src="img/code/hellonodejs_dev_branch.png" width=800/>
+
+The `UI Dev` gate of the value stream represents the `development` branch of the **helloapp-nodejs** app. The `Web Tests` stage tests whether the hosted nodejs app shows a body of `Hello World!`, spelled correctly. The current state of the app has an error (the body shows `Hello Worlld`) that we will fix in the next few steps. <p><img src="img/code/hellonodejs_failed_test.png" width=800/>
+
+### Fixing the **helloapp-nodejs** Application
+
+1. In DevOptics you should see that the **UI Dev** gate has failed due to the error mentioned above. <p><img src="img/code/failed_ui_dev.png" width=800/>
+2. Open the GitHub editor for the `hello.js` file in the **development** branch of your forked **helloapp-nodejs** repository, on line 13 fix the misspelled ***Worlld***   
+```
+res.render('index', { title: 'Hello', message: 'Hello World!', 
+```
+3. Commit the change to the **development** branch with the commit message ***UI-1001 fixed misspelling*** <p><img src="img/code/fix_ui_misspelling.png" width=800/>
+4. Now your job should complete successfully and the **UI Dev** gate should show that it finished successfully - what is your MTTR for the **UI Dev** gate? What is the MTTR for the entire Value Stream?  <p><img src="img/code/ui_dev_vs_success.png" width=800/>
+5. Now that you have fixed your **helloapp-nodejs** application it is time to merge to the **master** branch and deploy. Create a [Pull Request](https://help.github.com/en/articles/creating-a-pull-request) between the **development** branch and **master** branch of your forked **helloapp-nodejs** repository. 
+6. Changed the **base repository** to the **master** branch of your forked **helloapp-nodejs** repository (not the **cloudbees-dwjw** repository), add a comment and then click the **Create pull request** button <p><img src="img/code/ui_dev_to_master_pr.png" width=600/>
+7. A job will be created for the pull request and once it has completed successfully your pull request will show that **All checks have passed**. Go ahead and click the **Merge pull request** button and then click the **Confirm merge** button but DO NOT delete the **development** branch <p><img src="img/code/ui_merge_pr.png" width=600/>
+8. In your DevOptics Value Stream you should see the **UI-1001** ticket move from the **UI Dev** gate to the **UI Master** gate. What is your deployment frequency? <p><img src="img/code/final_microservice_vs.png" width=800/>
```

Comment:
Just curious why you spell out "deployment frequency" here but refer to MTTR above? Maybe at least put DF in parentheses.

Reply:
Not sure, that question was left over from baseline cb days content. 
Removed that, and replaced with a new screenshot focusing on value stream metrics.